include/flatbuffers/flatbuffers.h

```diff
@@ -960,9 +960,11 @@ FLATBUFFERS_FINAL_CLASS
   }
 
   /// @brief Store a string in the buffer, which can contain any binary data.
-  /// @param[in] str A const reference to a std::string to store in the buffer.
+  /// @param[in] str A const reference to a std::string like type with support
+  /// of T::c_str and T::length() to store in the buffer.
   /// @return Returns the offset in the buffer where the string starts.
-  Offset<String> CreateString(const std::string &str) {
+  template<typename T>
+  Offset<String> CreateString(const T &str) {
```

Comment:
I agree that having to change the generated code for this is undesirable. I guess I am worried about breaking any existing code out there, since we also have overloads for `const char *` etc, and some compilers may have trouble deciding which one to call. It also makes for hard to follow errors.

If C++ guarantees it will always prefer the overloads before it tries the template, then we can merge this. In fact if that is the case we can even leave the std::string overload in.